example_packages/unet2d/setup.py

```diff
@@ -0,0 +1,27 @@
+"""Install package."""
+from pathlib import Path
+
+import setuptools
+
+PROJECT_DIR = Path(__file__).parent.resolve()
+LONG_DESCR = (PROJECT_DIR / "README.md").read_text(encoding="utf-8")
+REQUIRES = ["imageio", "numpy", "pyyaml", "torch", "tqdm"]
+
+setuptools.setup(
```

Comment:
Currently, example packages can only be loaded through the engine, I think it would be a bit misleading to add a setup.py file. Otherwise, we need to figure out a way to actually run the package alone.

Reply:
I'm planning to add support for entry_points via setup.py file:
https://setuptools.readthedocs.io/en/latest/setuptools.html#dynamic-discovery-of-services-and-plugins
https://packaging.python.org/guides/creating-and-discovering-plugins/#using-package-metadata

We can use it in the engine to install packages.

Besides that, it's good to be able to install the example package and have imports work correctly.

Reply:
OK, interesting idea, let's try that then. We just need to make it optional since the engine may not have the permission to install package, or one may simply don't want to write a setup.py file.

Reply:
Yes, it should be optional and it will fit it the config yaml here:
```yaml
requirements:
  conda:
    - python=3.6.8
    - pytorch=1.1.0
  pip:
    - pyyaml
    - PACKAGE_WITH_ENTRYPOINT
```
We'll search for those entrypoints and load them if they exist.

Reply:
So you mean we can then remove the entrypoint definition in the yaml file? What if multiple package with entry point exists? Perhaps we still need to specify which module in yaml.

I thought you were talking about discover installed model packages from the engine.

Reply:
Yes, if we have a standard for what the entry point function should be called we could just call that function in the discovered package.

Each entry point package should have at least one entry point according to what we decide. We can decide the standard that should be followed.

Yes the discovery can be done separately. But we also need to install the package somehow.